simphony_ui/liggghts_wrapper_creation.py

```diff
@@ -69,3 +69,54 @@ def create_liggghts_wrapper(liggghts_settings):
     ]
 
     return liggghts_wrapper
+
+
+def create_liggghts_datasets(liggghts_wrapper, liggghts_settings):
+    """ Creates the liggghts particles datasets from the settings as provided
+    by the model object
+
+    Parameters
+    ----------
+    liggghts_wrapper : LiggghtsWrapper
+        Your liggghts wrapper in which you want to put the datasets
+    liggghts_settings : LiggghtsModel
+        The traited model describing the liggghts parameters
+
+    Returns
+    -------
+    flow_dataset :
+        A dataset containing flow particles
+    wall_dataset :
+        A dataset containing wall particles
+    """
+    input_file = liggghts_settings.input_file
+
+    list_particles = liggghts.read_data_file(input_file)
+
+    flow_particles = list_particles[0]
+    wall_particles = list_particles[1]
+
+    flow_particles.name = 'flow_particles'
+    wall_particles.name = 'wall_particles'
+
+    # Shift box_origin to (0,0,0) and update particles accordingly
+    box_origin = \
+        flow_particles.data_extension[liggghts.CUBAExtension.BOX_ORIGIN]
+
+    for particle in flow_particles.iter_particles():
+        particle.coordinates = (
+            particle.coordinates[0] - box_origin[0],
+            particle.coordinates[1] - box_origin[1],
+            particle.coordinates[2] - box_origin[2])
+        flow_particles.update_particles([particle])
+
+    flow_particles.data_extension[liggghts.CUBAExtension.BOX_ORIGIN] = \
+        (0.0, 0.0, 0.0)
+
+    liggghts_wrapper.add_dataset(flow_particles)
```

Comment:
I would be tempted to keep this in the client code. The fact is, this routine has two roles:
- it returns something it creates, in a strange way
- it modifies a passed argument (liggghts_wrapper).

which means that it has a side effect. Side effects on the arguments can create problems. I would be tempted to extract the code that just creates the datasets, then return those, then let the caller combine the result of this routine with the wrapper. This would probably result in not having to pass liggghts_wrapper at all.